R/class-workbook.R

```diff
@@ -6970,12 +6974,23 @@ wbWorkbook <- R6::R6Class(
         self$app$Company <- xml_node_create("Company", xml_children = company)
       }
 
-      xml_properties[core_created] <- xml_node_create(core_created,
-        xml_attributes = c(
-          `xsi:type` = "dcterms:W3CDTF"
-        ),
-        xml_children = format(as_POSIXct_utc(datetime_created), "%Y-%m-%dT%H:%M:%SZ")
-      )
+      if (!is.null(datetime_created)){
+        xml_properties[core_created] <- xml_node_create(core_created,
+          xml_attributes = c(
+            `xsi:type` = "dcterms:W3CDTF"
+          ),
+          xml_children = format(as_POSIXct_utc(datetime_created), "%Y-%m-%dT%H:%M:%SZ")
+        )
+      }
```

Comment:
https://github.com/JanMarvin/openxlsx2/blob/406c55f4feb09f3cb086e4e752944894c0bb3999/R/class-workbook.R#L348-L349

In the `initialize` function the assertion makes sure that there is an initial date set (NULL is not allowed). `set_property` picks up the core values of the workbook in 

https://github.com/JanMarvin/openxlsx2/blob/406c55f4feb09f3cb086e4e752944894c0bb3999/R/class-workbook.R#L6928-L6946

so that the create date is retained, it is not omitted when the `datetime_create=` argument has its new standard value `NULL` in `set_properties`.

```r
wb <- wb_workbook()
wb$set_properties(datetime_create = NULL)
wb$get_properties()["datetime_created"]
##       datetime_created 
## "2024-11-13T07:47:16Z" 
## 
```

So a workbook created by `wb_workbook` should always have a create date, I think? The alternative is a workbook that is read in from an xlsx file, which should also be valid. Is there another way to create a workbook or another way to modify properties than by `set_properties`?


Microsoft Excel 2013 (Office 15) happily accepts modify date < create date:

![image](https://github.com/user-attachments/assets/836fd401-9133-4d33-aa5a-268d6ad81c6f)

Reply:
Perfect ! Thanks for the explanation